server/eclipse-project/src/main/java/us/freeandfair/corla/hibernate/Persistence.java

```diff
@@ -181,148 +195,250 @@ private static synchronized void setupSessionFactory() {
   }
 
   /**
-   * Saves the specified object as an entity in the current session.
+   * @return true if a long-lived transaction is running in this thread, 
+   * false otherwise.
+   * @exception PersistenceException if the database isn't running.
+   */
+  public static boolean isTransactionRunning() 
+      throws PersistenceException {
+    if (hasDB()) {
+      final Pair<Session, Transaction> session_transaction = transaction_info.get();
+
+      return session_transaction != null && 
```

Comment:
This logic is repeated (essentially) on a bunch of lines. Can we factor it out into a method or function whose name would also summarize the conditions being checked?